comictalker/talkers/comicvine.py

```diff
@@ -163,10 +165,74 @@ class CVIssueDetailResults(TypedDict, total=False):
 CV_RATE_LIMIT_STATUS = 107
 
 
+def comicvine_settings(parser: settngs.Manager) -> None:
+    # Comic Vine settings
+    parser.add_setting(
+        "--series-match-search-thresh",
+        default=90,
+        type=int,
+    )
+    parser.add_setting("--use-series-start-as-volume", default=False, action=argparse.BooleanOptionalAction)
+    parser.add_setting(
+        "--clear-metadata",
+        default=True,
+        help="Clears all existing metadata during import, default is to merges metadata.\nMay be used in conjunction with -o, -f and -m.\n\n",
+        dest="clear_metadata_on_import",
+        action=argparse.BooleanOptionalAction,
+    )
+    parser.add_setting(
+        "--remove-html-tables",
+        default=False,
+        action=argparse.BooleanOptionalAction,
+        help="Removes html tables instead of converting them to text",
+    )
+    parser.add_setting(
+        "--cv-api-key",
+        help="Use the given Comic Vine API Key (persisted in settings).",
+    )
+    parser.add_setting(
+        "--cv-url",
+        help="Use the given Comic Vine URL (persisted in settings).",
+    )
+    parser.add_setting(
+        "-a",
+        "--auto-imprint",
+        action=argparse.BooleanOptionalAction,
+        default=False,
+        help="Enables the auto imprint functionality.\ne.g. if the publisher is set to 'vertigo' it\nwill be updated to 'DC Comics' and the imprint\nproperty will be set to 'Vertigo'.\n\n",
+    )
+
+    parser.add_setting(
+        "--sort-series-by-year", default=True, action=argparse.BooleanOptionalAction, help="Sorts series by year"
+    )
+    parser.add_setting(
+        "--exact-series-matches-first",
+        default=True,
+        action=argparse.BooleanOptionalAction,
+        help="Puts series that are an exact match at the top of the list",
+    )
+    parser.add_setting(
+        "--always-use-publisher-filter",
+        default=False,
+        action=argparse.BooleanOptionalAction,
+        help="Enables the publisher filter",
+    )
+    parser.add_setting(
+        "--clear-form-before-populating-from-cv",
+        default=False,
+        action=argparse.BooleanOptionalAction,
+        help="Clears all existing metadata when applying metadata from ComicVine",
+    )
+
+
 class ComicVineTalker(ComicTalker):
     default_api_key = "27431e6787042105bd3e47e169a624521f89f3a4"
     default_api_url = "https://comicvine.gamespot.com/api"
 
+    @staticmethod
+    def register_settings(parser: settngs.Manager) -> None:
+        parser.add_group("comicvine", comicvine_settings, False)
```

Comment:
I did consider some of the options being general, as I did last but I think it makes sense to keep them unique to each talker. With one talker you may want to sort by year but not another for example. Even with the publisher filter. Which of the settings do you think should be general to all talkers?

Reply:
Never referenced in comictaggerlib except in relation to setting the value:
* **--series-match-search-thresh**
* **--use-series-start-as-volume**
* **--remove-html-tables**
* **--cv-api-key**
* **--cv-url**

Only referenced in comictaggerlib:
* **--clear-metadata**: Does not make sense to send to the Talker, the Talker never sees any metadata to override
* **--auto-imprint**: Could be used but now depends on loading external data and if a Talker does not implement it it won't be there, leaving it in comictaggerlib ensures that it can always be used
* **--sort-series-by-year**: Makes sense as a potential option but if a Talker does not implement it it won't be there, leaving it in comictaggerlib ensures that it can always be used. Instead of moving this to the Talker I would rather have an option to have comictaggerlib not sort the returned data
* **--exact-series-matches-first**: Same as above
* **--always-use-publisher-filter**: Could be used but if a Talker does not implement it it won't be there, leaving it in comictaggerlib ensures that it can always be used
* **--clear-form-before-populating-from-cv**: Does not make sense to send to the Talker, it is a library and does not interact with any forms

It would make sense to make it so that some of these can be set per Talker but that would add additional complexity that I do not want to deal with. Primarily it would duplicate the settings to each talker, which in turn would mean that you could tune precisely what happens for several different sources, but what happens when a new source is added, what settings should we use? What would a user that just downloaded ComicTagger and used ComicVine, the default source, and configured it to do what they want expect when they add a new source? For the technically savvy, they are probably using it in a script which means that for their primary usage they can update their script to use different command-line parameters for each source, and if it's a file only option, which there aren't many left, they can use the --config option to point to a different folder. Yes it means that it is not a great experience if they want a different filter for each source, but there is also nothing stopping a source from implementing their own filter option, e.g. `--mu-publisher-filter`, the only real restriction on what types can be used, is that if it is a file settings it must be a type that the default json [encoder] and [decoder] can process.

Also note that since comictalker is a separate library I intend as much as possible for nothing to be added to it that would force it to be used with ComicTagger it should be something that can be extracted and used in a separate project (we'll see how well this actually turns out)

[decoder]: https://docs.python.org/3/library/json.html#json.JSONDecoder
[encoder]: https://docs.python.org/3/library/json.html#json.JSONEncoder

Reply:
I've split them out in https://github.com/comictagger/comictagger/pull/403/commits/d36337edbcafcde3b7f722563485915349d81beb may still want to move some around.

I now have the `talkerapi` adding the group settings after being called from `main` `register_options` but that may not be what you intend either?

With the constructor and settings is the idea to pass in `options[0]` and then have the talker parse out it's own settings?

Reply:
No, like [flake8], there would be a parse_options function that I would send a dictionary with all of the plugin's options.
As an example this is what would be currently sent to ComicVineTalker on d36337edbcafcde3b7f722563485915349d81beb
```python
{
    "series_match_search_thresh": 90,
    "use_series_start_as_volume": False,
    "remove_html_tables": False,
    "cv_api_key": "",
    "cv_url": "",
}
```

[flake8]: https://flake8.pycqa.org/en/latest/plugin-development/plugin-parameters.html#accessing-parsed-options